app/front-end/src/features/editor/components/hierarchy.tsx

```diff
@@ -1,3 +1,267 @@
+  marginBottom: theme.spacing(0.5),
+  marginTop: theme.spacing(0.5),
+  padding: theme.spacing(0.5),
+  paddingRight: theme.spacing(1),
+  fontWeight: 500,
+  ['&.Mui-expanded ']: {
+    '&:not(.Mui-focused, .Mui-selected, .Mui-selected.Mui-focused) .labelIcon': {
+      color: theme.palette.mode === 'light' ? theme.palette.primary.main : theme.palette.primary.dark,
+    },
+    '&::before': {
+      content: '""',
+      display: 'block',
+      position: 'absolute',
+      left: '16px',
+      top: '44px',
+      height: 'calc(100% - 48px)',
+      width: '1.5px',
+      backgroundColor: theme.palette.mode === 'light' ? theme.palette.grey[300] : theme.palette.grey[700],
+    },
+  },
+  '&:hover': {
+    backgroundColor: alpha(theme.palette.primary.main, 0.1),
+    color: theme.palette.mode === 'light' ? theme.palette.primary.main : 'white',
+  },
+  ['&.Mui-focused, &.Mui-selected, &.Mui-selected.Mui-focused']: {
+    backgroundColor: theme.palette.mode === 'light' ? theme.palette.primary.main : theme.palette.primary.dark,
+    color: theme.palette.primary.contrastText,
+  },
+}));
+
+const AnimatedCollapse = animated(Collapse);
+
+function TransitionComponent(props: TransitionProps) {
+  const style = useSpring({
+    to: {
+      opacity: props.in ? 1 : 0,
+      transform: `translate3d(0,${props.in ? 0 : 20}px,0)`,
+    },
+  });
+
+  return <AnimatedCollapse style={style} {...props} />;
+}
+
+const StyledTreeItemLabelText = styled(Typography)({
+  color: 'inherit',
+  fontSize: '1rem',
+  fontWeight: 500,
+  maxWidth: '100%',
+  wordWrap: 'break-word',
+}) as unknown as typeof Typography;
+
+interface CustomLabelProps {
+  children: React.ReactNode;
+  icon?: React.ElementType;
+  expandable?: boolean;
+}
+
+function CustomLabel({ icon: Icon, expandable, children, ...other }: CustomLabelProps) {
+  return (
+    <TreeItem2Label
+      {...other}
+      sx={{
+        display: 'flex',
+        alignItems: 'center',
+      }}
+    >
+      {Icon && <Box component={Icon} className='labelIcon' color='inherit' sx={{ mr: 1, fontSize: '1.2rem' }} />}
+
+      <StyledTreeItemLabelText>{children}</StyledTreeItemLabelText>
+      {expandable && <DotIcon />}
+    </TreeItem2Label>
+  );
+}
+
+const isExpandable = (reactChildren: React.ReactNode) => {
+  if (Array.isArray(reactChildren)) {
+    return reactChildren.length > 0 && reactChildren.some(isExpandable);
+  }
+  return Boolean(reactChildren);
+};
+
+const getIconFromFileType = (fileType: FileType) => {
+  switch (fileType) {
+    case 'csv':
+      return ArticleIcon;
+    case 'txt':
+      return ArticleIcon;
+    case 'doc':
+      return ArticleIcon;
+    case 'folder':
+      return FolderRounded;
+    default:
+      return InsertDriveFileIcon;
+  }
+};
+
+interface CustomTreeItemProps
+  extends Omit<UseTreeItem2Parameters, 'rootRef'>,
+    Omit<React.HTMLAttributes<HTMLLIElement>, 'onFocus'> {}
+
+const CustomTreeItem = React.forwardRef(function CustomTreeItem(
```

Comment:
Might need to restructure more. Add folder `features/editos/components/hierarchy` add new file `customTreeItem`. If component uses new component, it should have it's own file.